org.gnome.SimpleScan.json

```diff
@@ -0,0 +1,180 @@
+{
+    "app-id": "org.gnome.SimpleScan",
+    "branch": "stable",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "42",
+    "sdk": "org.gnome.Sdk",
+    "rename-desktop-file": "simple-scan.desktop",
+    "rename-appdata-file": "simple-scan.appdata.xml",
+    "command": "simple-scan",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
```

Comment:
```suggestion
        "--device=all",
```

Reply:
I'm not sure it's a good idea to release it with this against developers statement.

BTW, on all computers I maintain (mom, sister, mine), never used anything else than network scanners with Simple Scan.

Reply:
It's been more than two years since [this comment](https://gitlab.gnome.org/GNOME/simple-scan/-/issues/21#note_739110) and there was not any progress since. So, I believe that the `--device=all` permission is still better than no Simple Scan Flatpak on Flathub.

Reply:
I am not sure why the developer is so against the `--device=all` permission, but does not have any problem with regular non-sandboxed (deb/rpm) packages.

Reply:
there is this part too: https://gitlab.gnome.org/GNOME/simple-scan/-/issues/21#note_848661

But shouldn't be possible to add closed source backends as extensions ?

Reply:
Aren't proprietary backends used only by some specific scanners?

Reply:
Sure :)

Reply:
I contacted simple-scan maintainer by mail about this issue.

Reply:
for the proprietary backends, there is another scanner software, for KDE, that uses sane and has the same question. It could be an opportunity to have this setup as a more generic extension. Some Epson require theirs (the V600 comes to mind)

Reply:
> BTW, on all computers I maintain (mom, sister, mine), never used anything else than network scanners with Simple Scan.

Lucky you.

Reply:
HP printers I guess...

Reply:
Hi, I'm the one who actually wrote the "please don't try to publish it on Flathub." sentence in the README (obviously with the maintainer validating and merging it), I can explain why that's something I inserted.

There's proprietary drivers stuff of course, though I would not consider this a blocker for now (but that's one of the disadvantages of having sane in the sandbox). The sane build as I did it is incomplete, as some dependencies are missing (net-snmp/libsnmp at least IIRC, but see https://gitlab.gnome.org/GNOME/simple-scan/-/issues/21#note_1101055 too), so IMO the sane flatpak build should have more backends and options enabled, to make sure that there's as much working scanners as possible with the Simple Scan flatpak. Of course ideally we'd have a "scanner portal" (just to passthrough a fd for direct communication to sane or something like dbus) but it's not so much a big deal, even if it'd be a nice to have. --device=all is kinda problematic because it gives access to other devices, but unless someone works on sane and xdp to make a portal, it's still better than no decent scanner on Flathub. There's a case in point where adding Simple Scan as-is would pretty much reduce incentive to actually make a more secure scanner access, with a portal. Non networked scanners is IMO something very much necessary (I myself have a handheld one which has just a USB connection). Putting information like "USB scanners not supported in this flatpak" in the app's appdata description or in this readme is very likely to not be seen at all most of the time, and it would very much likely yield complaints or bugs upstream.

So overall I'm not opposed to have Simple Scan on Flathub, but those reasons should be considered and fixed when possible, to ship it in a better state.

Reply:
Don't majority (if not all) of scanners on market support airscan these days? Maybe the drivers aren't the problem anymore.

Reply:
Most high quality scanners don't.

Reply:
Name one.

Reply:
Epson V600 (epson-kowa proprietary driver), V850 Pro. the older V700 (supported by open source driver)

That's 3. And I haven't gone through the whole product line.
Nor some other vendors.

Also I have some older Canon LIDE, inexpesive, perfectly functionnal.

Reply:
Canon LIDE is listed as compatible on sane-airscan [frontpage](https://github.com/alexpevzner/sane-airscan). I don't know about epson but do we really need 100% of scanners to be supported in order to publish this app?

Reply:
which model (it's a produt line not a single SKU)? Sure the two I have here aren't.

Reply:
They list Canon LiDE 300 and Canon LiDE 400. Are you sure yours doesn't support escl or wsd?

Reply:
they use the plustek or genesys drivers from sane.

So apparently naming 3 wasn't enough so now we are at 5. 

And I have an old Nikon CoolScan III (transparency) that works over SCSI and has a driver for SANE (unless it's broken, had it back then)

Reply:
It doesn't matter what driver from sane they use as long as they support escl or wsd protocols.